engine/engine.go

```diff
@@ -357,9 +357,16 @@ func (r *ruleEngine) createViolation(conditionResponse ConditionResponse, rule R
 		if r.incidentLimit != 0 && len(incidents) == r.incidentLimit {
 			break
 		}
+		lineNumber, hasLineNumber := m.Variables["lineNumber"]
```

Comment:
I think that is reasonable to update the contract of the providers to include this as a top level. I would look 
https://github.com/konveyor/analyzer-lsp/blob/main/provider/provider.go#L117 to start.